client/app/scripts/superdesk-workspace/workspace.js

```diff
@@ -0,0 +1,238 @@
+
+(function() {
+    'use strict';
+
+    angular.module('superdesk.workspace', [])
+        .service('workspaces', WorkspaceService)
+        .directive('sdDeskDropdown', WorkspaceDropdownDirective)
+        .directive('sdEditWorkspace', EditWorkspaceDirective)
+        ;
+
+    WorkspaceService.$inject = ['api', 'desks', 'session', 'preferencesService', '$q'];
+    function WorkspaceService(api, desks, session, preferences, $q) {
+
+        var PREFERENCE_KEY = 'workspace:active',
+            RESOURCE = 'workspaces',
+            deferEdit,
+            vm = this;
+
+        this.active = null;
+        this.edited = null;
+        this.create = create;
+        this.save = save;
+        this.cancelEdit = cancelEdit;
+        this.setActive = setActiveWorkspace;
+        this.setActiveDesk = setActiveDesk;
+        this.getActive = getActiveWorkspace;
+        this.getActiveId = getActiveWorkspaceId;
+        this.queryUserWorkspaces = queryUserWorkspaces;
```

Comment:
I don't like aliases so I'm using it sparingly ;)
those vars should probably go below this public api declaration so it might not be obvious what vm is.